### PR TITLE
feat: hardcoded identity transfers in strategy tests

### PR DESCRIPTION
**Pull request by pauldelucia**
## Issue being fixed or feature implemented

Changed `StartIdentities` `hard_coded` field so that the `StateTransition` is optional. This way we can easily use identities that already exist in the chain.

```
pub struct StartIdentities {
    pub number_of_identities: u16,
    pub keys_per_identity: u8,
    pub starting_balances: u64,
    pub extra_keys: KeyMaps,
    pub hard_coded: Vec<(Identity, Option<StateTransition>)>,
}
```

Also added handling for credit transfers where the sender and recipient are hardcoded, whereas before they were randomly selected from the known identities each time. This included optionally passing a new struct IdentityTransferInfo to OperationType::IdentityTransfer which specifies sender, recipient, and amount.

## How Has This Been Tested?

platform-tui

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
	- Introduced new test cases for identity operations, including transfers and document handling.
	- Added a new structure, `IdentityTransferInfo`, to encapsulate identity transfer details.
	- Enhanced the flexibility of identity transfer operations by allowing specification of sender and recipient identities.
	- Updated identity insertion strategies to improve control over identity management during tests.

- **Bug Fixes**
	- Enhanced error handling and assertions in the testing framework for better validation of operations.

- **Refactor**
	- Modified existing tests and structures to improve data handling and enhance flexibility in identity management.
	- Updated method signatures to improve code clarity and maintainability.
	- Improved the handling of identity state transitions and dependencies in the code structure.

### CODE REVIEW

File: packages/strategy-tests/src/transitions.rs [hunk 1]
```diff
@@ -824,8 +824,8 @@ pub fn create_identity_credit_transfer_transition(
     let identity_public_key = identity
         .get_first_public_key_matching(
             Purpose::TRANSFER,
-            HashSet::from([SecurityLevel::CRITICAL]),
-            HashSet::from([KeyType::ECDSA_SECP256K1, KeyType::BLS12_381]),
+            SecurityLevel::full_range().into(),
+            KeyType::all_key_types().into(),
```
Comment (QuantumExplorer) [inline packages/strategy-tests/src/transitions.rs:828]: I think this change will not work.

Comment (pauldelucia) [reply packages/strategy-tests/src/transitions.rs:828]: Oops I was trying to debug something with qwizzie and forgot to revert that

File: packages/strategy-tests/src/lib.rs [hunk 2]
```diff
@@ -1287,38 +1288,65 @@ impl Strategy {
...
+
+                                // Use the sender index on the original slice
+                                let sender = &mut current_identities[random_index_sender];
+
+                                let state_transition = create_identity_credit_transfer_transition(
                                     sender,
                                     recipient,
                                     identity_nonce_counter,
                                     signer,
                                     300000,
```
[the remark this answers is not in the document]
Comment (pauldelucia) [reply packages/strategy-tests/src/lib.rs:1346]: You can if you select IdentityTransferSpecific. I _can_ add that for IdentityTransferRandom as well but I didn't yet

File: packages/strategy-tests/src/transitions.rs (hunk 1 above)
Comment (pauldelucia) [reply packages/strategy-tests/src/transitions.rs]: Fixed

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**